pkg/cgrouprate/cgrouprate.go

```diff
@@ -280,27 +299,76 @@ func Check(kube *processapi.MsgK8s, ktime uint64) {
 		name:  string(kube.Docker[:]),
 	}
 
-	handle.ch <- cq
+	glSt.handle.ch <- cq
 	cgroupratemetrics.CgroupRateTotalInc(cgroupratemetrics.Check)
 }
 
 func Config() {
-	if handle == nil {
+	if glSt.handle == nil {
 		return
 	}
 
-	if cgroupRateOptionsMap.MapHandle == nil {
-		handle.log.Warn("failed to update cgroup rate options map")
+	if glSt.cgRateOptionsMap.MapHandle == nil {
+		glSt.handle.log.Warn("failed to update cgroup rate options map")
 		return
 	}
 
 	key := uint32(0)
 	opts := processapi.CgroupRateOptions{
-		Events:   handle.opts.Events,
-		Interval: handle.opts.Interval,
+		Events:   glSt.handle.opts.Events,
+		Interval: glSt.handle.opts.Interval,
 	}
 
-	if err := cgroupRateOptionsMap.MapHandle.Put(key, opts); err != nil {
+	if err := glSt.cgRateOptionsMap.MapHandle.Put(key, opts); err != nil {
 		cgroupratemetrics.CgroupRateTotalInc(cgroupratemetrics.UpdateFail)
 	}
 }
+
+func RegisterCgroupRate(sensor *sensors.Sensor) (*sensors.Sensor, error) {
+	if !option.CgroupRateEnabled() {
+		return sensor, nil
+	}
+
+	glSt.mu.Lock()
+	defer glSt.mu.Unlock()
+
+	if glSt.handle != nil {
+		return nil, fmt.Errorf("cgrouprate: handle is already set, need to cleanup first")
+	}
+
+	var rateProgs []*program.Program
+	var optsProgs []*program.Program
+	for _, p := range sensor.Progs {
+		if base.IsExecve(p) || base.IsFork(p) || base.IsExit(p) {
+			rateProgs = append(rateProgs, p)
+		}
+		if base.IsExecve(p) {
+			optsProgs = append(optsProgs, p)
+		}
+	}
+
+	if len(optsProgs) <= 0 {
```

Comment:
should we check both opts and rate? len(xxx) can't be < 0 right?

Reply:
Yap, will fix.